services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -80,6 +96,76 @@ def check_indexable(feature: FeatureType) -> None:
     return indexable_columns
 
 
+def compute_length_column(
+    parquet_directory: Path,
+    column_name: str,
+    target_df: Optional[pl.DataFrame],
+    dtype: LengthDtype,
+) -> pl.DataFrame:
+    column_class = ListColumn if dtype == "list" else StringColumn
+    df = pl.read_parquet(str(parquet_directory / "*.parquet"), columns=[column_name])
+    lengths_column_name = f"{column_name}.length"
+    lengths_df: pl.DataFrame = column_class.compute_transformed_data(
+        df, column_name, transformed_column_name=lengths_column_name
+    )
+    if target_df is None:
+        return lengths_df.select(pl.col(lengths_column_name))
+
+    target_df.insert_column(target_df.shape[1], lengths_df[lengths_column_name])
+    return target_df
+
+
+def compute_audio_duration_column(
+    parquet_directory: Path,
+    column_name: str,
+    target_df: Optional[pl.DataFrame],
+) -> pl.DataFrame:
+    duration_column_name = f"{column_name}.duration"
+    durations = AudioColumn.compute_transformed_data(parquet_directory, column_name, AudioColumn.get_duration)
+    duration_df = pl.from_dict({duration_column_name: durations})
+    if target_df is None:
+        return duration_df
+    target_df.insert_column(target_df.shape[1], duration_df[duration_column_name])
+    return target_df
+
+
+def compute_image_width_length_column(
```

Comment:
I imagine the idea was:

```suggestion
def compute_image_width_height_column(
```